staging/src/k8s.io/apimachinery/pkg/util/net/http.go

```diff
@@ -132,13 +133,56 @@ func SetTransportDefaults(t *http.Transport) *http.Transport {
 	if s := os.Getenv("DISABLE_HTTP2"); len(s) > 0 {
 		klog.Infof("HTTP2 has been explicitly disabled")
 	} else if allowsHTTP2(t) {
-		if err := http2.ConfigureTransport(t); err != nil {
+		if err := configureHTTP2Transport(t); err != nil {
 			klog.Warningf("Transport failed http2 configuration: %v", err)
 		}
 	}
 	return t
 }
 
+func readIdleTimeoutSeconds() int {
+	ret := 0
+	// User can set the readIdleTimeout to 0 to disable the HTTP/2
```

Comment:
It looks like it defaults to zero if the user doesn't set it, is that what you intended? It doesn't seem to match the comment.

Reply:
This was to test if pull-kubernetes-e2e-gce-large-performance would pass if this feature was disabled (see https://github.com/kubernetes/kubernetes/pull/95981#issuecomment-720680851). Reverted now.